### ISSUE TITLE
Inconsistent behavior of shareReplay() when a new subscriber subscribes after previous subscription to source observable has completed

### ISSUE THREAD

**Issue by xwk** [Reactive-Extensions/RxJS#758]
Hi,

I'm not sure it is a bug, but I am very confused by the inconsistent behavior of shareReplay() in terms of whether it will re-subscribe to the source observable when a new subscriber subscribes after previous subscription to source observable has completed. It's easier to explain with code.

Program 1
````
var Rx = require('rx');

function createObserver(tag) {
    return Rx.Observer.create(
        function (x) {
            console.log('Next: ' + tag + x);
        },
        function (err) {
            console.log('Error: ' + tag + err);
        },
        function () {
            console.log('Completed ' + tag);
        });
}

var shared = Rx.Observable.just(0)
    .doOnNext(function (x) {
        console.log('Side effect ' + x);
    })
    .doOnCompleted(function(){
        console.log('Src completed');
    })
    .shareReplay();


shared.subscribe(createObserver('ObserverA'));
shared.subscribe(createObserver('ObserverB'));

Rx.Observable
    .return(true)
    .delay(5000)
    .flatMap(shared)
    .subscribe(createObserver('ObserverC'));
````
Program 1 will generate following output
````
Side effect 0
Src completed
Next: ObserverA0
Completed ObserverA
Side effect 0
Src completed
Next: ObserverB0
Completed ObserverB
...5 seconds later...
Side effect 0
Src completed
Next: ObserverC0
Completed ObserverC
````
You can see the "side effect 0" is printed thrice, which means the .shareReplay() resubscribed to the source observable when subscriber B comes after subscribe A has completely consumed the source observerable as well as when subscriber C comes after subscriber B has completely consumed the source observerable.

Program 2 in the below differs from program 1 by just simply delay the source observable by 1 ms. 
````
var Rx = require('rx');

function createObserver(tag) {
    return Rx.Observer.create(
        function (x) {
            console.log('Next: ' + tag + x);
        },
        function (err) {
            console.log('Error: ' + tag + err);
        },
        function () {
            console.log('Completed ' + tag);
        });
}

var shared = Rx.Observable.just(0)
    .delay(1)
    .doOnNext(function (x) {
        console.log('Side effect ' + x);
    })
    .doOnCompleted(function(){
        console.log('Src completed');
    })
    .shareReplay();


shared.subscribe(createObserver('ObserverA'));
shared.subscribe(createObserver('ObserverB'));

Rx.Observable
    .return(true)
    .delay(5000)
    .flatMap(shared)
    .subscribe(createObserver('ObserverC'));
````
Output of the program 2 changes to 
````
Side effect 0
Next: ObserverA0
Next: ObserverB0
Src completed
Completed ObserverA
Completed ObserverB
...5 seconds later...
Next: ObserverC0
Completed ObserverC
````
Notice the "side effect 0" message appears only once. By the time ObserverC subscribes to the shared observable, previous subscription to src observable has well ended. Why shareReplay() does not resubscribe to src observable as it did in the program 1?

**Comment by paulpdaniels** [comment]
Check the latest version, I think this was fixed in 2.5.3

**Comment by xwk** [comment]
@paulpdaniels , thanks. Verified 2.5.3 has this fixed. Now both program 1 and program 2 will only see one "Side effect 0" message.
